protocol_generator/protocol_generator.cpp

```diff
@@ -0,0 +1,234 @@
+
+		// Losuj permutację
+		std::random_shuffle ( temp_protocol.begin(), temp_protocol.end() );
+
+		// Losuj 2 czasy
+		// 72000 kroków symulacji co 6 sekund
+		// zaczynamy w pierwszym kroku nr 1
+		// między dawkami 30min = 300 kroków
+		// czyli:
+		// 1.1 weź długość dawek. dawkki mogą zacząć się najpóźniej 
+		//     w 72000 - długość * 300 kroku = 72000 - koniec
+		// 1.2 losuj pierwszy termin dawki z 1..72000-koniec
+		// 1.3 dodaj do niego 300 i losuj kolejny termin dawki z 
+		//     (poprzedni + 300, 14400-koniec + 300)
+		// 1.4 itd, aż nie dojdziesz do końca długości wektorów
+
+		std::vector<int> random_time_1;
+		std::vector<int> random_time_2;
+
+		int time_length = 0;
+
+		while (time_length < temp_length) {
+			std::uniform_int_distribution<int> dist_time(1, 72000);
+			int temp_time = dist_time(rng);
+			bool good_time = true;
+			for (int t = 0; t < random_time_1.size(); ++t) {
+				if ((std::abs(random_time_1[t] - temp_time)) < 300) {
+					good_time = false;
+				}
+			}
+			if (good_time) {
+				random_time_1.push_back(temp_time);
+				time_length++;
+			}
+		}
+
+		time_length = 0;
+
+		while (time_length < temp_length) {
+			std::uniform_int_distribution<int> dist_time(1, 72000);
+			int temp_time = dist_time(rng);
+			bool good_time = true;
+			for (int t = 0; t < random_time_2.size(); ++t) {
+				if ((std::abs(random_time_2[t] - temp_time)) < 300) {
+					good_time = false;
+				}
+			}
+			if (good_time) {
+				random_time_2.push_back(temp_time);
+				time_length++;
+			}
+		}
+
+		std::cout << "Good times.\n";
+
+		std::sort(random_time_1.begin(), random_time_1.begin() + temp_length);
+		std::sort(random_time_2.begin(), random_time_2.begin() + temp_length);
+
+		// Zapisz jeśli nie było powtórek
+		if (protocol_set.find(temp_protocol) == protocol_set.end()) {
+			std::cout << "Zapis.\n";
+			protocol_set.insert(temp_protocol);
+			i++;
+			std::cout << i << "...\n";
+			for (std::vector<float>::const_iterator val = temp_protocol.begin(); 
+				val != temp_protocol.end(); ++val) {
+					t << *val << ' ';
+			}
+			t << '\n';
+			for (std::vector<int>::const_iterator val = random_time_1.begin(); 
+				val != random_time_1.end(); ++val) {
+					f << *val << ' ';
+					t << *val << ' ';
+			}
+			f << '\n'; 
+			t << '\n';
+			for (std::vector<int>::const_iterator val = random_time_2.begin(); 
+				val != random_time_2.end(); ++val) {
+					f << *val << ' ';
+					t << *val << ' ';
+			}
+			f << '\n'; 
+			t << '\n'; 
+			if (temp_length <= 5) { 
+				// Protocol chosen for the benchmarking:
+				// every 24h = every 14400 steps, for 5 days
+				int benchmark_protocol1 = 1;
+				int position = 0;
+				for (int benchmark_protocol1 = 1; 
+					(benchmark_protocol1 < 72000) && (position < temp_length); 
+					benchmark_protocol1 += 14400) {
+						t << benchmark_protocol1 << ' ';
+						f << benchmark_protocol1 << ' ';
+						position++;
+				}
+				f << '\n'; 
+				t << '\n'; 
+			}
+
+			if (temp_length <= 8) {

```

Comment:
Magiczne stałe powinny być zmiennymi const z nazwami, żeby było wiadomo co znaczą

Reply:
ta stała jest używana raz, dodatkowo pod spodem był zamieszczony komentarz tłumaczący czemu długość to maksymalnie 8.